packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -80,30 +121,52 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
             {label}
           </FormLabel>
         )}
-        <Input
-          id={id}
-          name={name}
-          className={cn(
-            // Allow the arrows step
-            step && "show-arrows",
-            className,
+        <div className="relative flex items-center">
+          <Input
+            id={id}
+            name={name}
+            className={className}
+            pattern={isBigInt ? regex.toString() : pattern?.toString()}
+            type={inputType}
+            min={minValue}
+            max={maxValue}
+            aria-valuemin={minValue}
+            aria-valuemax={maxValue}
+            aria-invalid={!!errors?.length}
+            onKeyDown={blockInvalidChar}
+            value={displayValue}
+            defaultValue={defaultValue}
+            onChange={onChange}
+            step={step}
```

Comment:
If the native steps are disabled you shouldn't pass it down to the input

Reply:
done